tests/cli/bootsys/test_mgmt_power.py

```diff
@@ -177,3 +183,230 @@ def test_mgmt_ncns_skip_prompt_power_off(self):
+                                                  self.ncn_shutdown_timeout, self.ipmi_timeout)
+
+        # Assert call for Ceph unmount on ncn-m001
+        self.mock_do_ceph_unmounts.assert_called_once_with(self.mock_ssh_client, 'ncn-m001')
+
+        expected_messages = [
+            'Shutting down worker NCNs: ncn-w001, ncn-w002',
+            'Waiting up to 1200 seconds for worker NCNs to shut down...',
+            'Shutting down manager NCNs: ncn-m002, ncn-m003',
+            'Waiting up to 1200 seconds for manager NCNs to shut down...',
+            'Freezing Ceph and shutting down storage NCNs: ncn-s001, ncn-s002',
+            'Ceph freeze completed successfully on storage NCNs.',
+            'Waiting up to 1200 seconds for storage NCNs to shut down...',
+            'Shutdown and power off of storage NCNs: ncn-s001, ncn-s002',
+            'Shutdown and power off of all management NCNs complete.'
+        ]
+        self.assertEqual(expected_messages, [record.message for record in cm.records])
+
+    def test_do_mgmt_shutdown_power_with_fatal_error(self):
+        """Test do_mgmt_shutdown_power when a fatal error occurs."""
+        self.mock_do_ceph_freeze.side_effect = FatalPlatformError('Ceph freeze failed')
+        with self.assertLogs(level=logging.ERROR) as cm:
+            with self.assertRaises(SystemExit):
+                do_mgmt_shutdown_power(self.username, self.password, set(),
+                                       self.ncn_shutdown_timeout, self.ipmi_timeout)
+
+        self.mock_get_and_verify_ncn_groups.assert_called_once_with({'ncn-m001'})
+        self.assertEqual(cm.records[-1].message, 'Failed to freeze Ceph on storage NCNs: Ceph freeze failed')
+
+    def test_set_next_boot_device_to_disk_success(self):
+        """Test that the function sets the boot device to disk successfully"""
+        mock_stdout = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout.channel.recv_exit_status.return_value = 0
+        mock_stdout.read.return_value = b"Boot0001* UEFI OS\nBoot0002* Other OS"
+
+        self.mock_ssh_client.exec_command.return_value = (None, mock_stdout, mock_stderr)
+
+        with self.assertLogs(level=logging.INFO) as cm:
+            ncns = ['ncn-w001', 'ncn-s001']
+            set_next_boot_device_to_disk(self.mock_ssh_client, ncns)
+
+        expected_connect_calls = [call('ncn-w001'), call('ncn-s001')]
+        self.mock_ssh_client.connect.assert_has_calls(expected_connect_calls, any_order=True)
+
+        expected_exec_calls = [
+            call('efibootmgr'),
+            call('efibootmgr -n 0001'),
+            call('efibootmgr'),
+            call('efibootmgr -n 0001')
+        ]
+        self.mock_ssh_client.exec_command.assert_has_calls(expected_exec_calls, any_order=True)
+
+        self.assertEqual(cm.records[0].message, f'Successfully set next boot device to disk (Boot0001) for ncn-w001')
```

Comment:
f-string is unnecessary here:

```suggestion
        self.assertEqual(cm.records[0].message, 'Successfully set next boot device to disk (Boot0001) for ncn-w001')
```